src/control-plane/cloudfront-s3-portal.ts

```diff
@@ -168,29 +170,63 @@ export class CloudFrontS3Portal extends Construct {
 
     portalBucket.addToResourcePolicy(portalBucketPolicyStatement);
 
-
     // upload static web assets
     this.bucketDeployment = new BucketDeployment(this, 'portal_deploy', {
-      sources: process.env.IS_SKIP_ASSET_BUNDLE === 'true' ? [Source.data('test', 'test')] : [
-        Source.asset(props.frontendProps.assetPath, {
-          bundling: {
-            image: props.frontendProps.dockerImage,
-            command: props.frontendProps.buildCommand,
-            user: props.frontendProps.user,
-            outputType: BundlingOutput.NOT_ARCHIVED,
-            environment: props.frontendProps.environment,
-          },
-          assetHash: props.frontendProps.assetHash ?? undefined,
-          assetHashType: props.frontendProps.assetHashType ?? AssetHashType.SOURCE,
-        }),
+      sources: [
+        this.getWebAssets(props, true),
+      ],
+      destinationBucket: portalBucket,
+      prune: false,
+      distribution: this.distribution,
+      distributionPaths: props.frontendProps.autoInvalidFilePaths,
+      cacheControl: [
+        CacheControl.maxAge(Duration.days(30)),
+        CacheControl.immutable(),
+      ],
+    });
+    this.htmlDeployment = new BucketDeployment(this, 'portal_html_deploy', {
+      sources: [
+        this.getWebAssets(props, false),
       ],
       destinationBucket: portalBucket,
       prune: false,
       distribution: this.distribution,
       distributionPaths: props.frontendProps.autoInvalidFilePaths,
+      cacheControl: [
+        CacheControl.maxAge(Duration.seconds(0)),
+      ],
     });
   }
 
+  private getWebAssets(props: CloudFrontS3PortalProps, excludeIndexHtml: boolean) {
+    if (process.env.IS_SKIP_ASSET_BUNDLE === 'true') {
+      return Source.data('test', 'test');
+    } else {
+      const commands = [...props.frontendProps.buildCommands];
+      if (excludeIndexHtml) {
+        commands.push('cd /asset-output');
+        commands.push('rm /asset-output/index.html');
+      } else {
+        commands.push('cd /asset-output');
+        commands.push('ls | grep -xv index.html | xargs rm -rf');
+      }
+      return Source.asset(props.frontendProps.assetPath, {
```

Comment:
How about sharing the same source but using [exclude](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3_deployment.BucketDeploymentProps.html#exclude) and `include` `index.html`?

Creating two sources probably slows down the build of `cdk synth`.

Reply:
Has no effect on Assets bundled using the `bundling` property.
Ref: https://github.com/aws/aws-cdk/blob/7c62d6809cbdeee251c8ff31e22db7d0ff85aba2/packages/aws-cdk-lib/core/lib/fs/options.ts#L64

Reply:
Thanks for sharing it.